src/relationship/relationship_hash.js

```diff
@@ -0,0 +1,99 @@
+import Ember from 'ember';
+
+/**
+ * Define HashMap class
+ *
+ * Simple use of associative array as hash table allowing buckets w/
+ * chaining for duplicates.
+ */
+
+var RelationshipNode = Ember.Object.extend({
+
+	next: undefined,
+	prev: undefined,
+	item: undefined,
+
+	initialize: Ember.on('init', function() {
+		this.setProperties({
+			next: undefined,
+			prev: undefined,
+			item: undefined
+		});
+	})
+
+});
+
+
+var RelationshipHash = Ember.Object.extend({
+
+	buckets: [],
```

Comment:
Unless I misunderstand the intent here, don't we want {} not []? It looks like the code is indexing via ids, which are strings. While an array would work, that's simply because it's an object. If you add array['foo'] = 'something', and then print array.length, it's still 0. So using an array miscommunicates how we're actually using this data structure. Furthermore, in the event an id is a number (or gets coerced as such), now we're going to grow the array by how many ever ids there are, which is not the intent.

Unless I'm missing something, change to {}